codes/oscillators/number_phase.yml

```diff
@@ -0,0 +1,64 @@
+#######################################################
+## This is a code entry in the error correction zoo. ##
+##       https://github.com/errorcorrectionzoo       ##
+#######################################################
+
+# code id, physical, logical are all lower case
+code_id: number_phase
+physical: oscillators
+logical: qubits
+
+# Apostrophes are denoted by two apostrophe characters, i.e., ''
+# Code title is singular
+name: 'Number phase'
+
+# First reference with the code (optional).
+introduced: '\cite{doi:10.1103/PhysRevX.10.011058}'
+
+# Anything applicable to a larger parent set of codes (see below) should go in
+# that entry instead of here.
+description: 'Number phase codes are unnormalized bosonic rotation codes that can be approximately implemented physically via a normalization procedure. The logical states of an order \(N\) number phase code are \(\vert 0_L \rangle = \sum_{m=0}^{2N-1} \vert m\pi/N \rangle\) and \(\vert 0_L \rangle = \sum_{m=0}^{2N-1} (-1)^m \vert m\pi/N \rangle\), where \(\vert m\pi/N \rangle = \sum_{n=0}^\infty \mathrm{e}^{\mathrm{i} n \phi} \vert n \rangle\). '
```

Comment:
First do exact construction, mention Pegg-Barnett phase states explicitly. Conclude with "Ideal codewords are not normalizable, so approximate versions have to be constructed."